src/funnel/FPS.cpp

```diff
@@ -0,0 +1,331 @@
+
+FPS is divided in two components (fps.lp and fps.ld) which evaluate the projection of the ligand along the funnel line
+and the distance from it, respectively. The values attributed to these two components are then used together with the
+potential file created by the \ref FUNNEL bias to define if the ligand is within or not in the funnel-shape restraint
+potential. In the latter case, the potential will force the ligand to enter within the funnel boundaries.
+
+\par Examples
+
+The following input tells plumed to print the FPS components for the COM of a ligand. The inputs are a reference structure,
+which is the structure used for the alignment, the COM of the molecule we want to track, and 2 points in the cartesian space
+(i.e., x, y, and z) to draw the line representing the funnel axis.
+\plumedfile
+lig: COM ATOMS=2446,2447,2448,2449,2451
+fps: FPS REFERENCE=protein.pdb LIGAND=lig POINTS=5.3478,-0.7278,2.4746,7.3785,6.7364,-9.3624
+PRINT ARG=fps.lp,fps.ld
+\endplumedfile
+
+It is recommended to add a line to force the reconstruction of the periodic boundary conditions. In the following example,
+\ref WHOLEMOLECULES was added to make sure that the ligand was reconstructed together with the protein. The list contains
+all the atoms reported in the start.pdb file followed by the ANCHOR atom and the ligand. All atoms should be contained in the
+same entity and the correct order.
+\plumedfile
+WHOLEMOLECULES ENTITY0=54,75,212,228,239,258,311,328,348,372,383,402,421,463,487,503,519,657,674,690,714,897,914,934,953,964,
+974,985,1007,1018,1037,1247,1264,1283,1302,1324,1689,1708,1727,1738,1962,1984,1994,2312,2329,2349,2467,2490,2500,2517,2524,2536,
+2547,2554,2569,2575,2591,2607,2635,2657,2676,2693,2700,2719,2735,2746,2770,2777,2788,2795,2805,2815,2832,2854,2868,2898,2904,
+2911,2927,2948,2962,2472,3221,3224,3225,3228,3229,3231,3233,3235,3237
+lig: COM ATOMS=3221,3224,3225,3228,3229,3231,3233,3235,3237
+fps: FPS LIGAND=lig REFERENCE=start.pdb ANCHOR=2472 POINTS=4.724,5.369,4.069,4.597,5.721,4.343
+PRINT ARG=fps.lp,fps.ld
+\endplumedfile
+
+*/
+//+ENDPLUMEDOC
+
+
+class FUNNEL_FPS : public Colvar {
+
+  // Those are arrays that I created for the colvar
+  std::vector<AtomNumber> ligand_com;
+  std::vector<AtomNumber> anchor;
+  std::vector<AtomNumber> numbers;
+  bool pbc;
+  PLMD::RMSD* alignment;
+  PLMD::PDB* pdb;
+  bool squared;
+private:
+  vector<double> points;
+public:
+  explicit FUNNEL_FPS(const ActionOptions&);
+// active methods:
+  virtual void calculate();
+  static void registerKeywords(Keywords& keys);
+// I need a method in RMSDCoreCalc and these were requested
+  std::vector<double> align;
+  std::vector<double> displace;
+// It is written no more desctructors, but an expert said it's necessary for imported variables (pdb and alignment) or else memory leak
+  ~FUNNEL_FPS();
+};
+
+using namespace std;
+
+PLUMED_REGISTER_ACTION(FUNNEL_FPS,"FPS")
```

Comment:
@h2nch2co2h you should replace the string `FPS` here with `FUNNEL_FPS` (and also in the regtests). We have this convention that either contributed modules define a single action (e.g. `EDS`) or if they define multiple actions they have all the same prefix (e.g. `VES_`). Thanks!

Reply:
Thanks again for your review. I am sorry, I thought that the flag to call the action could remain FPS. Is it the only change I should make?